runtime/commonMain/src/kotlinx/serialization/internal/NullableSerializer.kt

```diff
@@ -1,6 +1,7 @@
 /*
  * Copyright 2017-2019 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
  */
+@file:UseExperimental(InternalSerializationApi::class)
```

Comment:
It's a bit weird that we have to opt-in for experimental annotations on the file level.
Why won't we use "-Xuse-experimental=kotlinx.serialization.*` in all main source sets?

Reply:
For this particular annotation, it is, of course, more reasonable to enable it in build file for whole source set. `ImplicitReflectionSerializer` is not enabled because it imposes some restrictions on runtime (e.g. does not work on Native) and we, as library developers, want to make sure that it would not 'infect' big parts of the library, for users' convenience.